tests/hypothesis_strategies.py

```diff
@@ -0,0 +1,158 @@
+"""Hypothesis strategies."""
+import random
+
+import hypothesis.strategies as st
+import numpy as np
+from hypothesis import assume
+from hypothesis.extra.pandas import columns, data_frames
+
+from baybe.exceptions import NumericalUnderflowError
+from baybe.parameters.categorical import (
+    CategoricalEncoding,
+    CategoricalParameter,
+    TaskParameter,
+)
+from baybe.parameters.custom import CustomDiscreteParameter
+from baybe.parameters.numerical import (
+    NumericalContinuousParameter,
+    NumericalDiscreteParameter,
+)
+from baybe.parameters.substance import SubstanceEncoding, SubstanceParameter
+from baybe.utils import DTypeFloatNumpy
+
+_largest_lower_interval = np.nextafter(
+    np.nextafter(np.inf, 0, dtype=DTypeFloatNumpy), 0, dtype=DTypeFloatNumpy
+)
+"""
+The largest possible value for the lower end of a continuous interval such that there
+still exists a larger but finite number for the upper interval end.
+"""
+
+_decorrelate = st.one_of(
```

Comment:
Why is this strategy "private"?

Reply:
you asked me the same question already on Azure :D go and see if my answer there still makes sense to you. If not, let me know ;)